### DISCUSSION TITLE
Most suitable double precision type for floating point computations ?

### DISCUSSION THREAD

**Discussion by antoinecarme** [duckdb/duckdb#5016] [General]
### What happens?

Hi all,

(Following #4931)

I am working on an application that translates machine learning models into SQL code for deploying these in-database. Roughly speaking, this works by generating a huge/explicit SQL code that produces the same dataset as the native python code used to generate the model itself. I am trying to tweak it for duckdb, it already supports postgresql.

The SQL codes the application generates uses a lot of floating point computations and can reach some limit cases where small differences can be seen between hardware platforms and SQL dialects. Sometimes it generates +inf and -inf literal floats.

Many thanks to the author to this document :

https://duckdb.org/docs/sql/data_types/numeric

I read the document and I have some doubts that need to be clarified here :

1. Is there a differece between duckdb and postgres when dealing with floating points ? I see that Float type is not the same in the two cases ('SELECT 3.1415926535897932384626433e+307::float as a' executes Ok on postgresql and fails with a conversion error on duckdb).
2. Does duckdb support IEEE-754 ?
3. Do you recommend the good old DOUBLE ? It it the same as Float(53) ?

I need a floating type that is IEEE-754 compliant and that holds 64 bit floats. For most databases the answer "DOUBLE or FLOAT" is OK.

Thanks in advance,

Antoine

### To Reproduce

Not a bug. Support question.

### OS:

Debian GNU Linux

### DuckDB Version:

'0.5.1', '7c111322d'

### DuckDB Client:

Python

### Full Name:

Antoine CARME

### Affiliation:

AC

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by hannes** [comment]
`DOUBLE` is what you're looking for I believe

**Comment by antoinecarme** [comment]
@hannes,

Thanks a lot for the quick answer.
